leveldb/leveldb.go

```diff
@@ -255,6 +259,25 @@ func (s *DB) Close() error {
 	return nil
 }
 
+// Clear will destroy the database and create it again
+func (s *DB) Clear() error {
+	err := s.Close()
```

Comment:
can you check if the batch handle needs to be restarted?
I would suggest to extract a method from new to call here as well if it is the case.

Reply:
Agree, the simple way I see is to reset the batch after opening the DB again in order to avoid nested locks that otherwise could give ugly deadlocks